### DISCUSSION TITLE
Possible Arc Welder bug

### DISCUSSION THREAD

**Discussion by Troublemakerprint** [FormerLurker/ArcWelderPlugin#173] [Q&A]
So I've been running Arc Welder both the octoprint plugin and the cura plugin (seperatly) for a little bit now and have noticed a "strange" issue. It seems after some layers the print head decides to trace the perimeter of the print bed and then resume the next layer.

Unsure if anyone else has had this issue.

**Comment by seantapscott** [comment]
This happened to me on an Ender 5, and when I reviewed communication logs,
it was declaring that the input buffer that I had set up on the serial USB
connection was not large enough to fit the larger commands of the arcs.  I
changed the input buffer size from 63 bytes to 127 bytes, and the error
messages went away.

So we can help further, could you also verify your firmware type, release
version number, and printer?

**Reply by Troublemakerprint** [reply]
Im running the Marlin 2.0 on the bugfix branch for my Tronxy X5SA (https://github.com/rhapsodyv/Marlin).

**Reply by FormerLurker** [reply]
What version of arcwelder are you using @Troublemakerprint?  The buffer overrun issue has been solved by turning off dynamic resolution by default.  You can scan your start/end gcode for code with large numbers of decimals (like 1.001000 or something).  If there are too many decimals and the command is large, gcode buffers can be overrun.  For example:  ```G1 X102.00234 Y123.456790 I123.456790 J123.456790 F4000.00000```

**Reply by Troublemakerprint** [reply]
I'm currently running 1.00. had a quick look at a print that i know was doing it and no sign of massive decimals.

**Comment by FormerLurker** [comment]
@Troublemakerprint, thanks for getting back.  Was not available over the weekend, so sorry for the delay.

I have heard of this issue, and it definitely seems to be firmware related.  You might try the beta version to see if anything changes, since there have been a lot of changes.  There is another discussion here that has instructions if you are interested.  I am in the process of creating a development printer, designed to let me swap out boards and firmware, so hopefully I will have the capability of actually debugging something like this on actual hardware.  I'll keep you posted.
